src/components/legend/defaults.js

```diff
@@ -58,6 +56,8 @@ module.exports = function legendDefaults(layoutIn, layoutOut, fullData) {
 
     if(showLegend === false) return;
 
+    layoutOut.legend = containerOut;
```

Comment:
legend's visibility is controlled by `showlegend`, which is *outside* the `legend` container itself, unlike most of our newer containers that have `visible` *inside* the container. Unless and until we change that (v2?) the legend container should not exist when it's not needed. This was causing me problems as `relinkPrivateKeys` was humorously removing `newFullLayout.legend` if it was empty and `oldFullLayout.legend` existed... so its existence would toggle on `Plotly.react` calls, making it look like the legend was always in need of updating 💫

Reply:
Very nice catch. 



> until we change that (v2?)

I'd vote +1 here.

Reply:
added to #420

Reply:
> the legend container should not exist when it's not needed. 

so by default, in _fullLayout, the .legend key wouldn't be there?
we read from _fullLayout to display widgets in the editor, if there's no .legend key, or a ._fullLayout.legend.visible key, then the widget won't display..

We could have a work around, but it's kind of one of the basic premises of widget visibility in the editor..

Reply:
@VeraZab you must not have been dependent on the existence of a `legend` container up to now, as it was unreliable ^^ - `showlegend` is still there, but now it's *reliably* the case that if `_fullLayout.showlegend=false` there will be NO `_fullLayout.legend` container.